src/main/java/com/taskadapter/redmineapi/RedmineManagerFactory.java

```diff
@@ -202,13 +202,33 @@ public static ClientConnectionManager createInsecureConnectionManager() {
      * Creates a connection manager with extended trust relations. It would 
      * use both default system trusted certificates as well as all certificates
      * defined in the <code>trustStores</code>.
-     * @param trustStores list of additional trust stores.
+     * @param trustStores list of additional trust stores to be included in the
+     *           trust chain. The server will be validated against all system-provided
+     *           CAs and all the ones provided via this list.
      * @return connection manager with extended trust relationship.
      */
     public static ClientConnectionManager createConnectionManagerWithExtraTrust(Collection<KeyStore> trustStores) throws KeyManagementException, KeyStoreException {
     	return createConnectionManager(BetterSSLFactory.createSocketFactory(trustStores));
     }
 
+        /**
+     * Creates a connection manager with extended trust relations and Client 
+     * Certificate authentication. It would use both default system trusted 
+     * certificates as well as all certificates defined in the 
+     * <code>trustStores</code>.
+     * @param keyStore key store containing the client certificate to use.
+     * @param keyStorePassword the keyStore password string.
+     * @param trustStores list of additional trust stores to be included in the
+     *           trust chain. The server will be validated against all system-provided
+     *           CAs and all the ones provided via this list.
+     * @return connection manager with extended trust relationship.
+     * @throws KeyManagementException
+     * @throws KeyStoreException
+     */
+    public static ClientConnectionManager createConnectionManagerWithClientCertificate(KeyStore keyStore, String keyStorePassword, Collection<KeyStore> trustStores) throws KeyManagementException, KeyStoreException {
```

Comment:
I am not a big fan of super-strict formatting rules, but still having a super long line makes it harder to review :)

Reply:
My bad, sorry! I'll split it when I address README.md